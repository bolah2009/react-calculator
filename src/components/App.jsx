import React from 'react';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import styles from '../css/modules/App.module.css';
import calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
      currentButton: null,
    };
  }

  handleClick(buttonName) {
    const newCalculations = calculate(this.state, buttonName);
    this.setState({ ...newCalculations, currentButton: buttonName });
  }

  render() {
    const { next, total, currentButton } = this.state;
    return (
      <div className={`d-flex col ${styles.calculator}`}>
        <Display currentButton={currentButton} result={next || total} />
        <ButtonPanel clickHandler={i => this.handleClick(i)} />
      </div>
    );
  }
}

export default App;
