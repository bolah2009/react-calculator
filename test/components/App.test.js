import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../src/components/App';

describe('App component should ', () => {
  test('should render BurronPanel and Display components', () => {
    const app = renderer.create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  });
});
