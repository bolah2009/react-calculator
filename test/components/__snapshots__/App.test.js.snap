// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should  should render ButtonPanel and Display components 1`] = `
<div
  className="d-flex col calculator"
>
  <div
    className="d-flex col jc-c display"
  >
    <div
      className="currentButton"
    />
    <div
      className="mainDisplay"
      id="display"
    >
      0
    </div>
  </div>
  <div
    className="d-flex as-c button-group group-one"
  >
    <button
      className="button"
      id="button-AC"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      id="button-+/-"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      id="button-%"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button"
      id="button-รท"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="d-flex as-c button-group group-two"
  >
    <button
      className="button"
      id="button-7"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button"
      id="button-8"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button"
      id="button-9"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button"
      id="button-X"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="d-flex as-c button-group group-three"
  >
    <button
      className="button"
      id="button-4"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button"
      id="button-5"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button"
      id="button-6"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button"
      id="button--"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="d-flex as-c button-group group-four"
  >
    <button
      className="button"
      id="button-1"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button"
      id="button-2"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button"
      id="button-3"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button"
      id="button-+"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="d-flex as-c button-group group-five"
  >
    <button
      className="button wide"
      id="button-0"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button"
      id="button-."
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightgrey",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button"
      id="button-="
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
